processor/pom.xml

```diff
@@ -57,7 +57,12 @@
             <artifactId>mapstruct</artifactId>
             <scope>provided</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.eclipse.tycho</groupId>
```

Comment:
@agudian, one thing I am wondering: Did you actually test this in a running Eclipse? I.e. can the processor "see" the JDT classes in the OSGi environment?